app/models/manageiq/providers/openstack/infra_manager/host.rb

```diff
@@ -139,7 +139,14 @@ def parent_credentials?
 
   def refresh_openstack_services(ssu)
     openstack_status = ssu.shell_exec("systemctl -la --plain | awk '/openstack/ {gsub(/ +/, \" \"); gsub(\".service\", \":\"); gsub(\"not-found\",\"(disabled)\"); split($0,s,\" \"); print s[1],s[3],s[2]}' | sed \"s/ loaded//g\"")
+    openstack_containerized_status = ssu.shell_exec("docker ps --format 'table {{.Names}}\t{{.Status}}' | tail -n +2")
```

Comment:
Are there circumstances under which this can fail and give output other than what you expected?

Reply:
If docker in not installed or there are no containers  `shell_exec` should return empty string.